### PR TITLE
Feature/blog v2

### PR DESCRIPTION
**Pull request by Tguntenaar**
_Resolves #735_
_Resolves #665_
_Resolves #732_
_Resolves #734_
_Resolves #568_

[Preview](https://near.org/thomasguntenaar.near/widget/app?page=community&handle=webassemblymusic&tab=blogv2)
will only work as soon as frol has executed the command and blogv2 is added to the contract.

Left to do:
- [x] Css styling
- [x] All accepetance criteria
- [x] Writing tests
- [x] Deploy preview
- [x] Add blogv2 addon to devhub contract @frol 

```zsh
near contract call-function as-transaction devhub.near create_addon json-args '{"addon":{configurator_widget:"devhub.near/widget/devhub.entity.addon.blogv2.Configurator",description: "Create a blog for your community",icon: "bi bi-substack",id: "blogv2",title: "BlogV2",view_widget: "devhub.near/widget/devhub.entity.addon.blogv2.Viewer"}}' prepaid-gas '100.000 TeraGas' attached-deposit '0 NEAR' sign-as frol.near network-config main sign-with-keychain send
```

[video.webm](https://github.com/NEAR-DevHub/neardevhub-bos/assets/28901891/69209882-1ae7-4714-b136-8eedb6f3720e)

[video.webm](https://github.com/NEAR-DevHub/neardevhub-bos/assets/28901891/21d9bb96-4ef7-4a5e-8708-2890687f1fcc)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new blog page component with dynamic content rendering and editing options.
  - Added components for displaying blog posts with dynamic content loading.
  - Implemented a blog post settings component with save and cancel functionality.
  - Introduced a category dropdown for selecting predefined blog categories.
  - Added functionality for creating and editing blog content, including title, subtitle, description, author, date, and category.
  - Implemented form components for editing blog content using Markdown.
  - Added a configurator component for handling blog addon settings.

### CODE REVIEW

**Comment by Tguntenaar** [comment]
```js
const blogv2 = {
  configurator_widget:
    "devhub.near/widget/devhub.entity.addon.blogv2.Configurator",
  description: "Create a blog for your community",
  icon: "bi bi-substack",
  id: "blogv2",
  title: "Blog V2",
  view_widget: "devhub.near/widget/devhub.entity.addon.blogv2.Viewer",
};
```

File: src/devhub/entity/addon/blogv2/editor/content.jsx [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+                </div>
+                <div className="text-muted text-xs">{option.description}</div>
+              </div>
+            ))}
+          </div>
+        )}
+      </div>
+    </DropdownBtnContainer>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Tguntenaar) [reply src/devhub/entity/addon/blogv2/editor/content.jsx:404]: Already implemented

**Comment by Megha-Dev-19** [comment]
https://github.com/NEAR-DevHub/neardevhub-bos/assets/100185149/8fb2d871-d490-4eec-83be-cfc7f9931926

Also another thing: when I click on "New Blog Post", the 0 index is new blog but when I click on any existing blog, it disappears, as you can see in video too.

**Review by petersalomonsen — APPROVED** [review]
I think the changes are ok, but also the suggestions from coderabbitai are good. The method that is not implemented for example ( with only a console.log ), and also the suggestion about using a default value instead of re-assigning the value from props is good. The error handling suggestions are not critical in my opinion, as long as the user clearly can see that a page reload might be needed because of some resources failed to load.

So not much I want to add, other than consider implementing the suggestions from the coderabbitai bot.

File: src/devhub/entity/addon/blogv2/editor/provider.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        },
+      },
+    },
+  });
+};
+
+const handleOnSubmitSettings = (v) => {
+  // TODO 599 move to provider
+  console.log("Implement saving blog settings issue 599");
+};
```
[the remark this answers is not in the document]
Comment (Tguntenaar) [reply src/devhub/entity/addon/blogv2/editor/provider.jsx]: Done

**Pull request merged**